terraform-provider-prefect_v2.92.1.exe plugin crashed

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

<!--- Thank you for keeping this note for the community --->

### Terraform Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. Use prefect_* if all resources or data sources are affected. --->

* prefect_XXXXX

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```tf
# Copy-paste your Terraform configurations here.
#
# For large Terraform configs, please use a service like Dropbox and share a link to the ZIP file.
# For security, you can also encrypt the files using our GPG public key:
#    https://www.hashicorp.com/security
#
# If reproducing the bug involves modifying the config file (e.g., apply a config,
# change a value, apply the config again, see the bug), then please include both:
# * the version of the config before the change, and
# * the version of the config after the change.
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, run `terraform apply` with the environment variable `TF_LOG=DEBUG`. See the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html) for more information.
--->

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

### Expected Behavior

<!--- What should have happened? --->

### Actual Behavior

<!--- What actually happened? --->

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: authenticating as a user instead of a service account? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

<!---
For Prefect Cloud customers who are working with a dedicated Sales/Solutions Engineer, please share this issue with them to expedite the investigation/resolution of the issue
--->

-----

I am using Terraform to create Prefect deployments, with the Terraform state stored in an AWS S3 bucket. Initially, everything was working as expected, I was able to deploy flows and apply changes without any issues.

However, the next day when I reopened the project and ran terraform plan after making a small change, I encountered a provider crash error. To troubleshoot, I removed the schedule resources from the Terraform state, destroyed the deployments, and redeployed all flows from scratch.

After redeploying, the flows were created successfully. But when I made a minor change, such as updating the deployment name, the same plugin crash error occurred again.

Am I missing something in my setup, or is there something incorrect in how I am managing these Prefect deployments with Terraform?

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x1619cbc]

goroutine 2626 [running]:
github.com/prefecthq/terraform-provider-prefect/internal/provider/resources.copyScheduleModelToResourceModel(0x0, 0xc001026180)
        github.com/prefecthq/terraform-provider-prefect/internal/provider/resources/deployment_schedule.go:428 +0x1c
github.com/prefecthq/terraform-provider-prefect/internal/provider/resources.(*DeploymentScheduleResource).Read(0xc0014074d0, {0x1a6d378, 0xc0011d7dd0}, {{{{0x1a73410, 0xc0014ab980}, {0x17d3900, 0xc0014ab4d0}}, {0x1a75498, 0xc001d57130}}, 0x0, ...}, ...)
        github.com/prefecthq/terraform-provider-prefect/internal/provider/resources/deployment_schedule.go:324 +0x52e
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadResource(0xc0000ca308, {0x1a6d378, 0xc0011d7dd0}, 0xc001bc3090, 0xc00144b5d8)
        github.com/hashicorp/terraform-plugin-framework@v1.16.1/internal/fwserver/server_readresource.go:156 +0xbae
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ReadResource(0xc0000ca308, {0x1a6d378?, 0xc0011d7ce0?}, 0xc001bc2eb0)
        github.com/hashicorp/terraform-plugin-framework@v1.16.1/internal/proto6server/server_readresource.go:71 +0x592
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ReadResource(0xc0000a7900, {0x1a6d378?, 0xc0011d6780?}, 0xc001f8af50)
        github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov6/tf6server/server.go:863 +0x2f3
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ReadResource_Handler({0x18de320, 0xc0000a7900}, {0x1a6d378, 0xc0011d6780}, 0xc000a83600, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:753 +0x1a9
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000b2200, {0x1a6d378, 0xc001405f20}, 0xc00251a480, 0xc0003455f0, 0x20e6a98, 0x0)
        google.golang.org/grpc@v1.75.1/server.go:1431 +0x1036
google.golang.org/grpc.(*Server).handleStream(0xc0000b2200, {0x1a6dc98, 0xc0000a0b60}, 0xc00251a480)
        google.golang.org/grpc@v1.75.1/server.go:1842 +0xb88
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.75.1/server.go:1061 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 13      
        google.golang.org/grpc@v1.75.1/server.go:1072 +0x11d

Error: The terraform-provider-prefect_v2.92.1.exe plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it      
can be fixed. The output above should help diagnose the issue.
```

--- a/internal/provider/resources/block.go
+++ b/internal/provider/resources/block.go
@@ -350,6 +350,8 @@ func (r *BlockResource) Read(ctx context.Context, req resource.ReadRequest, resp
 			"ID is unset",
 			"This is a bug in the Terraform provider. Please report it to the maintainers.",
 		)
+
+		return
 	}
 
 	client, err := r.client.BlockDocuments(state.AccountID.ValueUUID(), state.WorkspaceID.ValueUUID())

--- a/internal/provider/resources/deployment.go
+++ b/internal/provider/resources/deployment.go
@@ -706,6 +706,8 @@ func (r *DeploymentResource) Create(ctx context.Context, req resource.CreateRequ
 			"Error creating deployment client",
 			fmt.Sprintf("Could not create deployment client, unexpected error: %s. This is a bug in the provider, please report this to the maintainers.", err.Error()),
 		)
+
+		return
 	}
 
 	var tags []string
@@ -802,6 +804,8 @@ func (r *DeploymentResource) Read(ctx context.Context, req resource.ReadRequest,
 			"Error creating deployment client",
 			fmt.Sprintf("Could not create deployment client, unexpected error: %s. This is a bug in the provider, please report this to the maintainers.", err.Error()),
 		)
+
+		return
 	}
 
 	// A deployment can be imported + read by either ID or Handle
@@ -977,6 +981,8 @@ func (r *DeploymentResource) Delete(ctx context.Context, req resource.DeleteRequ
 			"Error creating deployment client",
 			fmt.Sprintf("Could not create deployment client, unexpected error: %s. This is a bug in the provider, please report this to the maintainers.", err.Error()),
 		)
+
+		return
 	}
 
 	deploymentID, err := uuid.Parse(state.ID.ValueString())

--- a/internal/provider/resources/deployment_schedule.go
+++ b/internal/provider/resources/deployment_schedule.go
@@ -318,9 +318,7 @@ func (r *DeploymentScheduleResource) Read(ctx context.Context, req resource.Read
 
 	schedule, err := getScheduleByID(state, schedules)
 	if err != nil {
-		// If the specific schedule is not found in the list, it was likely
-		// deleted externally. Remove it from state so Terraform can recreate it.
-		resp.State.RemoveResource(ctx)
+		resp.Diagnostics.AddError("Unable to get schedule by ID", err.Error())
 
 		return
 	}

--- a/internal/provider/resources/deployment_schedule.go
+++ b/internal/provider/resources/deployment_schedule.go
@@ -273,6 +273,9 @@ func (r *DeploymentScheduleResource) Create(ctx context.Context, req resource.Cr
 	// Additionally, we couldn't use getResourceByID here because of a race condition:
 	// we'd need an ID in the state to compare against, which doesn't exist yet.
 	resp.Diagnostics.Append(copyScheduleModelToResourceModel(schedules[0], &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 
 	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
 	if resp.Diagnostics.HasError() {
@@ -324,6 +327,9 @@ func (r *DeploymentScheduleResource) Read(ctx context.Context, req resource.Read
 	}
 
 	resp.Diagnostics.Append(copyScheduleModelToResourceModel(schedule, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 
 	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
 	if resp.Diagnostics.HasError() {
@@ -394,6 +400,9 @@ func (r *DeploymentScheduleResource) Update(ctx context.Context, req resource.Up
 	}
 
 	resp.Diagnostics.Append(copyScheduleModelToResourceModel(schedule, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 
 	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
 	if resp.Diagnostics.HasError() {

--- a/internal/provider/resources/deployment_schedule.go
+++ b/internal/provider/resources/deployment_schedule.go
@@ -318,7 +318,11 @@ func (r *DeploymentScheduleResource) Read(ctx context.Context, req resource.Read
 
 	schedule, err := getScheduleByID(state, schedules)
 	if err != nil {
-		resp.Diagnostics.AddError("Unable to get schedule by ID", err.Error())
+		// If the specific schedule is not found in the list, it was likely
+		// deleted externally. Remove it from state so Terraform can recreate it.
+		resp.State.RemoveResource(ctx)
+
+		return
 	}
 
 	resp.Diagnostics.Append(copyScheduleModelToResourceModel(schedule, &state)...)
@@ -387,6 +391,8 @@ func (r *DeploymentScheduleResource) Update(ctx context.Context, req resource.Up
 	schedule, err := getScheduleByID(plan, schedules)
 	if err != nil {
 		resp.Diagnostics.AddError("Unable to get schedule by ID", err.Error())
+
+		return
 	}
 
 	resp.Diagnostics.Append(copyScheduleModelToResourceModel(schedule, &plan)...)

